public/app/features/annotations/editor_ctrl.ts

```diff
@@ -56,17 +56,19 @@ export class AnnotationsEditorCtrl {
     this.reset();
 
     this.onColorChange = this.onColorChange.bind(this);
-  }
 
-  async datasourceChanged() {
-    return (this.currentDatasource = await this.datasourceSrv.get(this.currentAnnotation.datasource));
+    $scope.$watch('ctrl.currentAnnotation.datasource', async (value: string) => {
```

Comment:
Moved this from handler to $watch as from my memory this was more idiomatic for angular when some variables were dependent on each other.

Reply:
not sure, always tried to stay away from manual watch expressions as much as possible and do things when actually needed rather then check for changes on every digest. Was there an other reason for this change?

Reply:
think the $scope.apply is enough to fix the bug

Reply:
Sure will change. I thought it makes more sense when you have a direct relationship between variables but it is long time since last time I worked with angular.